### PR TITLE
Recommend 5.23+ users to disable jmx

### PR DESCRIPTION
**Pull request by SeBBBe**
With the latest commit, JMX is no longer mandatory as long as discovery service v2 is enabled. As Neo4j discourages the use of JMX, update our documentation to be in line with this recommendation. Not sure which version this change will land in.

----

If you open a PR that needs to go into a current version, you need to *cherry-pick your commit from dev over to the current version branch*. Only then will the proper builds that generate html/pdf be run. But beware: Docs will be generated but not published automatically!

- [x] N/A - or - I have added the appropriate "cherry-pick-to" labels to this PR so I don't forget to do this later!

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-ops-manager-247.surge.sh

Updated pages:
- [Neo4j instance requirements](https://neo4j-docs-ops-manager-247.surge.sh/ops-manager/1.12/addition/instance-requirements)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-ops-manager-247.surge.sh

Updated pages:
- [Neo4j instance requirements](https://neo4j-docs-ops-manager-247.surge.sh/ops-manager/1.12/addition/instance-requirements)
- [Server installation](https://neo4j-docs-ops-manager-247.surge.sh/ops-manager/1.12/installation/server)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-ops-manager-247.surge.sh

Updated pages:
- [Neo4j instance requirements](https://neo4j-docs-ops-manager-247.surge.sh/ops-manager/1.12/addition/instance-requirements)

File: modules/ROOT/pages/addition/instance-requirements.adoc [hunk 1]
```diff
@@ -66,7 +66,12 @@ For the metrics collection feature to work correctly, these configuration settin
 
 ** `server.metrics.filter=*`
 
-** `server.metrics.jmx.enabled=true`
+** Discovery service v2 *or* `server.metrics.jmx.enabled=true` (JMX)
```
Comment (alexnb) [inline modules/ROOT/pages/addition/instance-requirements.adoc]:
```suggestion
** Discovery service v2 *or* `server.metrics.jmx.enabled=true` (JMX metrics)
```

File: modules/ROOT/pages/addition/instance-requirements.adoc [hunk 2]
```diff
@@ -66,7 +66,12 @@ For the metrics collection feature to work correctly, these configuration settin
...
+
+[NOTE]
+====
+For Neo4j versions prior to 5.23, enabling link:https://neo4j.com/docs/java-reference/current/jmx-metrics/[JMX] is a requirement. For versions 5.23 or newer, it is recommended to leave JMX disabled and instead use link:https://neo4j.com/docs/operations-manual/current/clustering/setup/discovery/[discovery service v2].
```
Comment (alexnb) [inline modules/ROOT/pages/addition/instance-requirements.adoc]:
```suggestion
For Neo4j versions prior to 5.23, enabling link:https://neo4j.com/docs/java-reference/current/jmx-metrics/[JMX metrics] is a requirement.
For versions 5.23 or newer, it is recommended to leave JMX metrics disabled and instead use link:https://neo4j.com/docs/operations-manual/current/clustering/setup/discovery/[discovery service v2].
```

**Review by alexnb — APPROVED** [review]
LGTM, suggesting "every sentence on new line" and using "JMX metrics" to be more precise.

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Pull request merged**